docs/DeveloperGuide.md

```diff
@@ -1,34 +1,357 @@
+Targeted towards developers who are or want to work on CheatLogs.
+
+### 1.2. Product Scope
+
+This document will cover the high-level architecture of the program, as well as the details for the implementation of the features.
+
+#### 1.2.1. Target User Profile
 
 {Describe the target user profile}
 
-### Value proposition
+#### 1.2.2. Value Proposition
 
 {Describe the value proposition: what problem does it solve?}
 
-## User Stories
+#### 1.3. Design Goals
+
+Our main goal is to design a cheatsheet management system that will help novice and beginner programmers to learn and adapt to various programming languages. 
+Our cheatsheet app comes with a set of preloaded cheatsheets. 
+Additionally, the user can input their own cheat sheets into the application. 
+This will make cheatlogs a comprehensive, one-stop solution for every student to manage their cheatsheet.
+
+#### 1.4. Definitions
+
+
+
+## 2. Setting Up, Getting Started
+
+
+
+### 2.1. Prerequisites
+
+1. JDK 11.0.8
+2. Java IDE (IntelliJ Recommended)
+
+CheatLogs has been developed on Java 11 and may not be supported on other versions. 
+You are free to use any java file editor and run the program by following the step by step guide below. 
+
+### 2.2. Running the Project
+
+1. Ensure you have Java 11 or above installed in your Computer.
+2. Download the latest cheatlogs.jar from here.
+3. Move the file to a folder you want to use as the home folder for this application.
+4. Invoke java -jar cheatlogs.jar on the command line to run the program. A welcome message should appear, as shown below.
+
+![](Images/Image1.PNG)
+
+### 2.3 Importing into IntelliJ [optional]
+
+In case you prefer to use IDEs to develop. 
+IntelliJ IDEA community edition is a popular free choice. 
+After installing a version of it you can import it using the “Get from Version Control” option below and clone from our [repository](https://github.com/AY2021S1-CS2113T-W11-3/tp.).
+
+![](Images/Image2.PNG)
+
+## 3. Design
+
+This section will elaborate on the architecture and component design of CheatLogs.
+
+### 3.1. Architecture
+
+The image below illustrates the high-level design of CheatLogs.
+
+![](Images/Image3.PNG)
+
+Image 1: General Architecture of CheatLogs
+
+CheatLogs is split into 5 major components, each handling distinct features of the application. The components and a brief description of them is listed below.
+* `UI`: The user interface of the app.
+* `CheatSheet`: The structure of each cheat sheet
+* `CheatSheetList`: A collection of every cheat sheet.
+* `Parser`: Builds a data structure based on user inputs.
+* `Command`: An encapsulation of data and methods to execute each command
+* `Storage`: Updates application data based on relevant external files.
+
+The UML diagram below illustrates an extensive version of the various classes present in CheatLogs as well as their interactions with each other.
+
+![](Images/Image4.PNG)
```

Comment:
![image](https://user-images.githubusercontent.com/28430109/97657414-be06af80-1aa4-11eb-843d-3313a37ea2ca.png)
It would be good to include association labels and multiplicity.
And for some of the dotted lines, are you indicating that there is association or dependency? Because if it is a dependency then there should be an arrow